quickstarts/microsoft.network/vnet-two-subnets/main.bicep

```diff
@@ -0,0 +1,48 @@
+@description('VNet name')
+param vnetName string = 'VNet1'
+
+@description('Address prefix')
+param vnetAddressPrefix string = '10.0.0.0/16'
+
+@description('Subnet 1 Prefix')
+param subnet1Prefix string = '10.0.0.0/24'
+
+@description('Subnet 1 Name')
+param subnet1Name string = 'Subnet1'
+
+@description('Subnet 2 Prefix')
+param subnet2Prefix string = '10.0.1.0/24'
+
+@description('Subnet 2 Name')
+param subnet2Name string = 'Subnet2'
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+resource vnet 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: vnetName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+  }
+}
+
+resource subnet1 'Microsoft.Network/virtualNetworks/subnets@2020-06-01' = {
+  parent: vnet
+  name: subnet1Name
+  properties: {
+    addressPrefix: subnet1Prefix
+  }
+}
+
+resource subnet2 'Microsoft.Network/virtualNetworks/subnets@2020-06-01' = {
+  parent: vnet
+  name: subnet2Name
+  properties: {
+    addressPrefix: subnet2Prefix
+  }
+}
```

Comment:
I believe the decompiler placed a dependence on subnet1 here, which seems strange, but in another sample I see that dependence explicitly in the JSON between subnets.

Do subnets need to have a particular creation order?  Is a creation order guaranteed when subnets are listed under the vn instead of top-level?